### PR TITLE
6 get api/review/review id/comments

### PR DESCRIPTION
**Pull request by kontsio1**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -29,29 +30,70 @@ describe("1. GET /api/categories", () => {
...
-              category: expect.any(String),
-              review_img_url: expect.any(String),
-              created_at: expect.any(String),
-              votes: expect.any(Number),
-              designer: expect.any(String),
-              comment_count: expect.any(Number),
-            })
-          );
-        });
+    .get("/api/reviews/3/comments")
```
Comment (lewispricey) [inline __tests__/app.test.js:63]: Please can you add a test that the comments are ordered by the created date (most recent first) and an extra test that the user gets an empty array of comments when they request the comments for a valid ID that has no comments.

File: app.js [hunk 2]
```diff
@@ -1,9 +1,35 @@
...
+
+app.use((err, req, res, next) => {
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Very Bad Request!" });
+  } else {
+    next(err);
+  }
+});
+
+app.use((err, req, res, next) => {
```
Comment (lewispricey) [inline app.js:23]: It would be good to move the error handler functions out to a seperate file to keep the app.js clear

File: controllers/controller.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-const { selectCategories, selectReviews } = require("../models/model");
+const { selectCategories, selectReviews, selectComments } = require("../models/model");
```
Comment (lewispricey) [inline controllers/controller.js:1]: Great  consistent naming convention for your model & controller names! You may want to look at moving the models/controlers out to their own seperate files as the project grows to keep the files easily maintainable.

File: models/model.js [hunk 4]
```diff
@@ -24,3 +24,15 @@ exports.selectReviews = () => {
...
 };
+
+exports.selectComments = (review_id) => {
+  return db
+  .query(`SELECT * FROM comments WHERE review_id = $1 ORDER BY created_at DESC`, [review_id])
+  .then((data) => {
+    if (data.rows.length !== 0) {
+      return data.rows;
+    } else {
+        return Promise.reject({ msg: "Not Valid Review Id", status: 404 });
```
Comment (lewispricey) [inline models/model.js:35]: 404 is great for an ID that doesn't exist but what would happen if it the ID exists but has no comments?

**Review by lewispricey — CHANGES_REQUESTED** [review]
Great work so far! Really solid naming conventions and error handling.

I've commented with a few extra tests that would be worthwhile adding in and it would be great to see those error functions moved out to their own seperate file.

Please raise another PR once you've made the changes :)

File: __tests__/app.test.js [hunk 5]
```diff
@@ -29,29 +30,80 @@ describe("1. GET /api/categories", () => {
...
+// });
+
+describe("6. GET /api/comments/:review_id/comments", ()=>{
+  test("status:200, responds with an array of objects", ()=>{
+    return request(app)
+    .get("/api/reviews/3/comments")
+    .expect(200)
+    .then(({ body }) => {
+      const { comments } = body;
+      expect(comments).toBeSortedBy('created_at', {descending: true})
```
Comment (lewispricey) [inline __tests__/app.test.js:67]: Awesome, I could see the logic was written in your DB query but great to test the sort aswell just to ensure its what we'd expect 😄

File: __tests__/app.test.js [hunk 6]
```diff
@@ -29,29 +30,80 @@ describe("1. GET /api/categories", () => {
...
+            created_at: expect.any(String),
+            review_id: expect.any(Number),
+            author: expect.any(String),
+            body: expect.any(String)
+          })
+        );
+      });
+    });
+  })
+  test("status:200, responds with an empty array when there are no comments", () => {
```
Comment (lewispricey) [inline __tests__/app.test.js:82]: Awesome!

File: models/model.js [hunk 7]
```diff
@@ -24,3 +23,29 @@ exports.selectReviews = () => {
...
+    )
+    .then((data) => {
+      if (data.rows.length !== 0) {
+        return data.rows;
+      } else {
+        return db
+          .query(`SELECT * FROM reviews WHERE review_id = $1`, [review_id])
+          .then((data) => {
+            if (data.rows.length === 0) {
+              return Promise.reject({
```
Comment (lewispricey) [inline models/model.js:41]: Really like how you've handled this logic in the model to keep the controller clear for just returning the responses.
